x/chainlink/keeper/keeper.go

```diff
@@ -437,3 +437,25 @@ func (k Keeper) FeedOwnershipTransfer(ctx sdk.Context, feedOwnershipTransfer *ty
 
 	return ctx.BlockHeight(), ctx.TxBytes(), nil
 }
+
+func (k Keeper) RequestNewRound(ctx sdk.Context, requestNewRound *types.MsgRequestNewRound) (int64, []byte, error) {
+	// might be possible that this method can throw so might be best to move this to the very end.
+	roundStore := ctx.KVStore(k.roundStoreKey)
+	currentLatestRoundId := k.GetLatestRoundId(ctx, requestNewRound.FeedId)
+	roundId := currentLatestRoundId + 1
+
+	// update the latest roundId of the current feedId
+	roundStore.Set(types.GetRoundIdKey(requestNewRound.GetFeedId()), i64tob(roundId))
+
+	// emit NewRoundData event
+	err := types.EmitEvent(&types.MsgNewRoundDataEvent{
+		FeedId:   requestNewRound.FeedId,
+		RoundId:  roundId,
+		FeedData: nil,
```

Comment:
hm.. 2 questions here. 

1) It probably should be a new event type instead of `MsgNewRoundData` cuz apparently there is not data when feed owner send a new round request. say `MsgNewRoundRequest`, once the data provider received this type event, they will fire a new round data submission which will trigger `MsgNewRoundData` with actual feed data.

2) same question in the previous conversation but it's been marked as resolved. Why `RequestNewRound` will need to set a new roundId into store? (L448)

Reply:
1. ok we can do that, but i dont think it needs to be a brand new event. I think it is fine to emit the event with empty feedData. This will just be up to the endpoint (chainlink node) to interpret a `nil` feedData field to be simply just a round id increment. IMO i think it'd be better to send a `nil` FeedData, but if you feel strongly otherwise, it wouldn't be a problem to add an extra message type. 

But if we do add it, will we fire 2 events for `addFeedData` because that does 2 things: increment roundId AND add feedata?

2. There are two stores. there is one that is in charge of storing the round id and one for the feed data. The roundId store will be incremented but no data will be pushed to the feedStore.